addons/mrp/views/stock_picking_views.xml

```diff
@@ -111,4 +111,15 @@
             </xpath>
         </field>
     </record>
+
+    <record id="view_picking_form_inherit_mrp" model="ir.ui.view">
+        <field name="name">stock.picking.form</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form"/>
+        <field name="arch" type="xml">
+            <field name="picking_type_id" position="attributes">
+                <attribute name="domain">[('code', '!=', 'mrp_operation')]</attribute>
```

Comment:
@svs-odoo you can make this domain directly in stock right? and write it on the model please, not the view